model/command_webhook.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+type CommandWebhook struct {
+	Id        string
+	CreateAt  int64
+	CommandId string
+	UserId    string
+	ChannelId string
+	RootId    string
+	ParentId  string
+	UseCount  int
+}
+
+const (
+	COMMAND_WEBHOOK_LIFETIME = 1000 * 60 * 30
+)
+
+func (o *CommandWebhook) PreSave() {
+	if o.Id == "" {
+		o.Id = NewId()
+	}
+
+	if o.CreateAt == 0 {
+		o.CreateAt = GetMillis()
+	}
+}
+
+func (o *CommandWebhook) IsValid() *AppError {
+	if len(o.Id) != 26 {
+		return NewLocAppError("CommandWebhook.IsValid", "model.command_hook.id.app_error", nil, "")
```

Comment:
Apologies for not being clear (and the late response). In this case the errors probably won't make it back to a response but I think it's a good idea to use `NewAppError` for two reasons:
1. We're trying to replace all uses of `NewAppLocError` so that we only have one error constructor
2. Many of the other IsValid functions for other models return to the http response and it's nice to just be consistent and set a bad request error code throughout